### PR TITLE
Add Gateway conformance tests to ambient

### LINKED ISSUE #44742: GAMMA conformance tests should run under ambient
GAMMA now has conformance tests. We only run with sidecars though

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
We run all the Gateway API core conformance tests for sidecars, but not yet for ambient. This PR adds runs those tests for ambient as well.

Fixes #44742

### CODE REVIEW

**Comment by keithmattix** [comment]
Let me bump the Gateway-api rc2 to confirm

**Comment by keithmattix** [comment]
@howardjohn this is ready for review now that the traffic splitting is working as expected

File: tests/integration/ambient/gateway_conformance_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// defined in sigs.k8s.io/gateway-api/conformance/base/manifests.yaml
+var conformanceNamespaces = []string{
+	"gateway-conformance-infra",
+	"gateway-conformance-app-backend",
+	"gateway-conformance-web-backend",
+}
+
+var skippedTests = map[string]string{
+	"HTTPRouteRedirectPath":          "spec is wrong, pending https://github.com/kubernetes-sigs/gateway-api/pull/1880",
+	"HTTPRouteRedirectHostAndStatus": "spec is wrong, pending https://github.com/kubernetes-sigs/gateway-api/pull/1880",
```
Comment (howardjohn) [inline tests/integration/ambient/gateway_conformance_test.go:53]: this is in 0.7.0

File: tests/integration/ambient/gateway_conformance_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	"MeshFrontendHostname":           "https://github.com/istio/istio/issues/44702",
```
Comment (howardjohn) [inline tests/integration/ambient/gateway_conformance_test.go:54]: this one should actually work with ambient

File: tests/integration/ambient/gateway_conformance_test.go (hunk 1 above)
Comment (keithmattix) [reply tests/integration/ambient/gateway_conformance_test.go:53]: Looks like all of the skipped tests are; let me remove the first 2 for both pilot and ambient and the second for just ambient

File: tests/integration/ambient/gateway_conformance_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			}
+			if rev := ctx.Settings().Revisions.Default(); rev != "" {
+				opts.NamespaceLabels = map[string]string{
+					"istio.io/rev": rev,
+				}
+			} else {
+				opts.NamespaceLabels = map[string]string{
+					"istio-injection": "enabled",
+				}
+			}
```
[the remark this answers is not in the document]
Comment (keithmattix) [reply tests/integration/ambient/gateway_conformance_test.go:97]: Oh wow good catch. Completely missed this

**Review by howardjohn — APPROVED** [review]

File: tests/integration/ambient/gateway_conformance_test.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+			// remove the dataplane mode label from the gateway-conformance-infra namespace
+			// so that the ingress gateway doesn't get captured
+			ns, err := gatewayConformanceInputs.Client.Kube().CoreV1().Namespaces().Get(context.Background(), "gateway-conformance-infra", metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			delete(ns.Labels, constants.DataplaneMode)
+			_, err = gatewayConformanceInputs.Client.Kube().CoreV1().Namespaces().Update(context.Background(), ns, metav1.UpdateOptions{})
```
[the remark this answers is not in the document]
Comment (keithmattix) [reply tests/integration/ambient/gateway_conformance_test.go:115]: Ah thank you; will add a hold until I do this

**Review by hanxiaop — APPROVED** [review]
lgtm

**Pull request merged**